### PR TITLE
test: update address-codec tests to typescript 

### PR DESCRIPTION
**Pull request by ckniffen**
## High Level Overview of Change

- Switch files to typescript
- Add missing types in test files
- Change how the codec functions were imported. They now count towards the "% Funcs" calculation, increasing it from 75% to 100%.
- Lint test files

### Type of Change

- [x] Refactor (non-breaking change that only restructures code)
- [x] Tests (You added tests for code that already exists, or your new feature included in this PR)

### CODE REVIEW

File: packages/ripple-address-codec/jest.config.js [hunk 1]
```diff
@@ -3,5 +3,6 @@ const base = require('../../jest.config.base.js')
 
 module.exports = {
   ...base,
-  displayName: 'ripple-address-codec',
+  roots: [...base.roots, '<rootDir>/test'],
+  displayName: 'ripple-binary-codec',
```
Comment (mvadari) [inline packages/ripple-address-codec/jest.config.js:7]:
```suggestion
  displayName: 'ripple-address-codec',
```

**Review by jonathanlei — APPROVED** [review]

File: packages/ripple-address-codec/test/index.test.ts [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
+import {
   classicAddressToXAddress,
   xAddressToClassicAddress,
   isValidXAddress,
   encodeXAddress,
-} = require('./index')
+} from '../src'
 
-const testCases = [
+const testCases: Array<[string, false | number, string, string]> = [
```
Comment (JST5000) [inline packages/ripple-address-codec/test/index.test.ts:8]: Can you make an interface for this type to explain what these types mean? (It seems they may be `classicAddress`, `tag` | false, `mainnetAddress`, `testnetAddress` judging from the code on line 147)

File: packages/ripple-binary-codec/HISTORY.md [hunk 3]
```diff
@@ -7,13 +7,12 @@
...
 * Remove `assert` dependency. If you were catching `AssertionError` you need to change to `Error`.
 
-
 ## 1.9.0 (2023-08-24)
 
 ### Added
 * Add AMM support [XLS-30](https://github.com/XRPLF/XRPL-Standards/discussions/78)
 * Updated to include latest updates to `definitions.json`.
-
+*
```
Comment (JST5000) [inline packages/ripple-binary-codec/HISTORY.md:15]: Should this `*` be deleted?

**Review by JST5000 — APPROVED** [review]
Logic looks good - just two comments on style / organization

**Pull request merged**